spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/transaction/ReactiveFirestoreTransactionManager.java

```diff
@@ -0,0 +1,189 @@
+								.setTransaction(transactionObject.getTransactionId()).build(), obs))
+				.then();
+	}
+
+	private FirestoreGrpc.FirestoreStub getRequiredDatabase() {
+		return this.firestore;
+	}
+
+	private static ReactiveFirestoreTransactionObject extractFirestoreTransaction(Object transaction) {
+
+		Assert.isInstanceOf(ReactiveFirestoreTransactionObject.class, transaction,
+				() -> String.format("Expected to find a %s but it turned out to be %s.",
+						ReactiveFirestoreTransactionObject.class,
+						transaction.getClass()));
+
+		return (ReactiveFirestoreTransactionObject) transaction;
+	}
+
+	private static ReactiveFirestoreTransactionObject extractFirestoreTransaction(GenericReactiveTransaction status) {
+
+		Assert.isInstanceOf(ReactiveFirestoreTransactionObject.class, status.getTransaction(),
+				() -> String.format("Expected to find a %s but it turned out to be %s.",
+						ReactiveFirestoreTransactionObject.class,
+						status.getTransaction().getClass()));
+
+		return (ReactiveFirestoreTransactionObject) status.getTransaction();
+	}
+
+	private Mono<ReactiveFirestoreResourceHolder> newResourceHolder(TransactionDefinition definition) {
+		TransactionOptions transactionOptions = definition.isReadOnly()
+				? TransactionOptions.newBuilder().setReadOnly(TransactionOptions.ReadOnly.newBuilder().build()).build()
+				: TransactionOptions.newBuilder().setReadWrite(TransactionOptions.ReadWrite.newBuilder().build())
+						.build();
+
+		BeginTransactionRequest beginTransactionRequest = BeginTransactionRequest.newBuilder()
+				.setOptions(transactionOptions)
+				.setDatabase(this.databasePath)
+				.build();
+		return ObservableReactiveUtil
+				.<BeginTransactionResponse>unaryCall(
+						obs -> this.firestore.beginTransaction(beginTransactionRequest, obs))
+				.map(beginTransactionResponse -> new ReactiveFirestoreResourceHolder(
+						beginTransactionResponse.getTransaction()));
+	}
+
+	/**
+	 * Firestore specific transaction object, representing a
+	 * {@link ReactiveFirestoreResourceHolder}. Used as transaction object by
+	 * {@link ReactiveFirestoreTransactionManager}.
+	 */
+	private class ReactiveFirestoreTransactionObject implements SmartTransactionObject {
+		private @Nullable ReactiveFirestoreResourceHolder resourceHolder;
+
+		ReactiveFirestoreTransactionObject(@Nullable ReactiveFirestoreResourceHolder resourceHolder) {
+			this.resourceHolder = resourceHolder;
+		}
+
+		@Nullable
+		public ByteString getTransactionId() {
+			return this.resourceHolder != null ? this.resourceHolder.getTransactionId() : null;
+		}
+
+		@Nullable
+		public ReactiveFirestoreResourceHolder getResourceHolder() {
+			return this.resourceHolder;
+		}
+
+		public void setResourceHolder(@Nullable ReactiveFirestoreResourceHolder resourceHolder) {
+			this.resourceHolder = resourceHolder;
+		}
+
+		@Override
+		public boolean isRollbackOnly() {
+			return false;
+		}
+
+		@Override
+		public void flush() {
+
```

Comment:
Add comment explaining why it does not apply?

Reply:
Note: Flushing is intended primarily for technologies that have a session cache, inspired by JPA.